### ISSUE TITLE
Recaf as a Java Agent

### ISSUE THREAD

**Issue by Col-E** [Col-E/Recaf#31]
It would be ***REALLY*** neat if recaf could be applied to a jvm as an agent and the edits would be applied via the instrumentation API.

It'd be sorta like [JavaSnoop](https://code.google.com/archive/p/javasnoop/).

So this would probably pass some arguments to the main method, which would replace the file menu with a *"refresh classes"* button or something similar. It would still behave largely the same as before with the file tree, as that can just be retrieved via the class path / agent launch args. 

Since it would have access to all the libraries I'd imagine there should be some sorta option to limit what is accessible so some operations of recaf aren't slowed to a halt.

**Comment by andylizi** [comment]
Awesome idea!

**Comment by Col-E** [comment]
Soon™️

**Comment by Col-E** [comment]
Pushed what I got so far in https://github.com/Col-E/Recaf/commit/ea8f632d555af987e7beaa6ec1b7b0b7f3893ff0. The ability to save edits in agent-mode is not implemented yet, but everything else seems to work!

**Comment by Col-E** [comment]
As of 7d9b37ee089a642eda13cb3e2df8614d8d5fba49 saving only modified classes works! It only updates classes if you edit the method code though *(add / remove opcodes)*. Due to the way the ASM tree API is structured I still don't think I'll easily be able to listen to subtle changes like changing an invokevirtual's `owner`. I'll add a way to manually register classes until I can figure something out.

**Comment by Col-E** [comment]
I noticed when attaching to jvm's with older ASM libs already loaded things still worked, but the version check threw an exception in `ClassNode` so in 20192d736e8ab25ef4c6b16e24bccbbf079bacb5 agent-mode will detect the loaded version and use it. If no version of ASM is in the JVM it'll just load the one recaf comes with, V6.

Also I tried the custom extension idea with 137219196b48e0ffa3cff1033c9bccd3fa2e709f *(I reverted everything and misspelled `truly`)* and that seems like a really inefficient solution. I guess I'll settle with what we got now *(Detection of new opcodes and manual marking)*
